### ISSUE TITLE
There are no devices registered in your account on the developer website. Plug in and select a device to have Xcode register it.

### ISSUE THREAD

**Issue by ajzimme** [jasarien/Provenance#500]
Your guide says to get a free Apple Developer account.
Ok. Did that.
Error says to register my Apple TV.
Ok. Sure. I'll go ahead and do that- OH have to be a paid developer to do that. Hm.. - 
But your guide says to get a free Apple Developer account.
I'm confused.

**Comment by jasarien** [comment]
Not sure what to say, it's worked for everyone else without a paid dev account. 

Are you sure you're using the latest version of the dev tools?

**Comment by ajzimme** [comment]
Yeah I downloaded all the updates.
Even worse. I finally successfully got it building, but it doesn't show up
on the Apple TV!

**Comment by marcelvoss** [comment]
That's pretty weird but are you sure you followed all the instructions? Did you get any error when you tried to install Provenance on your ATV? Your description is pretty vague which makes it very difficult to solve this without more information. It could be a signing issue, an issue with provisioning profiles etc.

**Comment by lolkthxbai** [comment]
@ajzimmie you got it successfully building but did it finish? You should get an alert that says "build succeeded" and a few seconds after that it will run Provenance.

**Comment by ajzimme** [comment]
Build succeeded- never shows on Apple TV. I gave up already.

**Comment by jasarien** [comment]
You are using the "build and run" option, and not just the "build" option, right? And the Apple TV is plugged in to the Mac while you build and run? 

I can't think of any reason that provenance wouldn't show up if those things are true and the build succeeds.

**Comment by ajzimme** [comment]
I was pressing the play button, which builds and runs it, yes.

And I did clean before as well.

And yes, of course the Apple TV was plugged in.

**Comment by jasarien** [comment]
And you were building to the apple TV, not the simulator? Sorry for the basic questions, but as I said, if the build succeeds and you're using build and run there's no reason why it wouldn't show up.

**Comment by jasarien** [comment]
You're certain there were no errors during installation? As in the build succeeded, but something went wrong while copying over to the apple TV?

**Comment by ajzimme** [comment]
It's weird cuz I rebuilt it successfully 3x.

I'll rerun it tonight just to show you.

Give me a few hours.

**Comment by orenwolf** [comment]
FWIW I just built today from my Sierra imac, following the wiki, to an ATV4, worked without a hitch - once I plugged in the apple tv, it was then selectable as a build target.

couple gotchas for me:

1 - Needed to plug in the appleTV and power it, I wasn't sure if it would use bus power or not
2 - the bundleID for topshelf needed to be the SAME as the bundleID for provenance, with a .extension of some sort on it. That wasn't clear in the docs
3 - It's not obvious how to "stop" Provenance once its built and running in XCode, but I finally found the menu option for it.

Otherwise, from a clean git repo and fresh xcode install, everything went swimmingly.

**Comment by lolkthxbai** [comment]
@ajzimme, you mentioned before that you would try again and record it. Did you get to record the issue or did it resolve itself?

**Comment by ajzimme** [comment]
Nah I just gave up dude, I don't care anymore.

**Comment by orenwolf** [comment]
@ajzimme To other things to try:

1 - When you plug in the appleTV, did you select it as a build target? I think that didn't happen automatically for me - instead, it just said "generic appletv"

2 - are you sure your cable is good? I know there's been an issue with USB-C cables that only work for charging but won't carry data.

Good luck!

**Comment by ajzimme** [comment]
Yeah I did select the Apple TV, and I would assume the cable is good, but that's always a possibility.

**Comment by ChocoChippy** [comment]
Hi, do I need to pay for a developer account in order to side load an app to my AppleTV?

**Comment by ajzimme** [comment]
Yea

**Comment by QuarterSwede** [comment]
You don't need a paid developer account to get the app running on the Apple TV. You'll have to resign the app every 7 days though. A paid account gets you TopShelf functionality and you only have to resign it once a year.

**Comment by ChocoChippy** [comment]
That's what I was afraid of... thanks for confirming this.  I really appreciate the help!

**Comment by jeremyeccles** [comment]
I had the same issue. To solve it, I had to go into Build Settings in the Xcode project and set the Development Team to be my team in two different places. It was set to a team name of 63497P68S6, which Xcode did not recognize. After making this change, it built and ran without a problem.

**Comment by buckeyered80** [comment]
I second jeremyeccles's notion. I had the same error about signing. When your build fails, go to the area on the left pane with the red stop signs. Double Click on the Top Shelf header. It should open a new window and take you to the area of the build fails with the red stop sign errors. Change the signer to your apple id. You might have to also change the bundle identifier to something else. I just replaced the .com with my name. There were 3 areas I had to do this by changing the selection tab on the top left to switch between Provenance, Provenance-TV and Provenance-Tests.
